File: middleNode.py
class Node:
    def __init__(self, data, next_null=None) -> None:
        self.data = data
        self.next = next_null
        
def middleNode(head) -> Node:
    slow = head
    fast = head
    while fast != None and fast.next != None:
        slow = slow.next
        fast = fast.next.next
    return slow
def printLinkedList(head) -> None:
    temp = head
    while temp != None:
        print(temp.data,end=" ")
        temp = temp.next
    print()

if __name__ == "__main__":
    head = Node(1)
    head.next = Node(3)
    head.next.next = Node(2)
    head.next.next.next = Node(4)
    head.next.next.next.next = Node(5)
    head.next.next.next.next.next = Node(6)
    
    print("The original Linked List is : ")
    printLinkedList(head)
    
    print("The middle of the linked List is : ")
    ans = middleNode(head)
    print(ans.data)
    
    
    
    

File: MiddleNode.java

/**
 * Node
 */
class Node {
    int data;
    Node next;
    Node(int data, Node next) {
        this.data = data;
        this.next = next;
    }
    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class MiddleNode{

    public static Node middleNodeFinder(Node head){
        Node slow = head;
        Node fast = head;

        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }
        return slow;
    }
    public static void printLinkedList(Node head) {
        Node temp = head;
        while (temp !=  null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
        System.out.println();
    }
    public static void main(String[] args) {
        Node head = new Node(1);
        head.next = new Node(3);
        head.next.next = new Node(2);
        head.next.next.next = new Node(4);
        head.next.next.next.next = new Node(5);


        System.out.println("The Original Linked List is ");
        printLinkedList(head);

        Node ans = middleNodeFinder(head);
        System.out.println("The middle of the linked list is " + ans.data);
    }
}